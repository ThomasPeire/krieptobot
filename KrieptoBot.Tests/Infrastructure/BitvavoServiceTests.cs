using System;
using System.Threading.Tasks;
using FluentAssertions;
using KrieptoBot.Infrastructure.Bitvavo.Services;
using KrieptoBot.Tests.Mocks.Bitvavo;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace KrieptoBot.Tests.Infrastructure
{
    public class BitvavoServiceTests
    {
        private readonly MockBitvavoApi _mockService = new();
        private BitvavoService _bitvavoService;

        [SetUp]
        public void Setup()
        {
            _mockService.InitData();
            _bitvavoService = new BitvavoService(_mockService);
        }

        [Test]
        public async Task GetBalance_ShouldReturn_Balance()
        {
            var result = await _bitvavoService.GetBalanceAsync();

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetCandles_ShouldReturn_Candles()
        {
            var result = await _bitvavoService.GetCandlesAsync("BTC-EUR");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetMarkets_ShouldReturn_Markets()
        {
            var result = await _bitvavoService.GetMarketsAsync();

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetMarket_ShouldReturn_Market()
        {
            var result = await _bitvavoService.GetMarketAsync("BTC-EUR");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetAssets_ShouldReturn_Assets()
        {
            var result = await _bitvavoService.GetAssetsAsync();

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetAsset_ShouldReturn_Asset()
        {
            var result = await _bitvavoService.GetAssetAsync("BTC");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetTrades_ShouldReturn_Trades()
        {
            var result = await _bitvavoService.GetTradesAsync("BTC-EUR");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetOrders_ShouldReturn_Orders()
        {
            var result = await _bitvavoService.GetOrdersAsync("BTC-EUR");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetOrder_ShouldReturn_Order()
        {
            var result =
                await _bitvavoService.GetOrderAsync("BTC-EUR", new Guid("4a7bd126-2d21-4918-96dc-0c8f51760a0b"));

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task GetOpenOrder_ShouldReturn_OpenOrder()
        {
            var result = await _bitvavoService.GetOpenOrderAsync("BTC-EUR");

            result.Should().MatchSnapshot();
        }

        [Test]
        public async Task PostOrder_ShouldReturn_AddOrder()
        {
            var order = await _bitvavoService.PostBuyOrderAsync("BTC-EUR", "Limit", 9000.00m, 0.0001m);

            var result = await _bitvavoService.GetOrderAsync(order.MarketName.Value, order.Id);

            Assert.That(order.MarketName.Value, Is.EqualTo(result.MarketName.Value));
            Assert.That(order.Side, Is.EqualTo(result.Side));
            Assert.That(order.Type, Is.EqualTo(result.Type));
            Assert.That(order.Amount, Is.EqualTo(result.Amount));
            Assert.That(order.Price, Is.EqualTo(result.Price));
        }
    }
}