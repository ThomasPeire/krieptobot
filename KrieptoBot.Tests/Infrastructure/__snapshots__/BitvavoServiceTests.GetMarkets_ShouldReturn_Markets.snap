[
  {
    "Name": "AAVE-EUR",
    "BaseSymbol": {
      "Value": "AAVE"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ADA-BTC",
    "BaseSymbol": {
      "Value": "ADA"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ADA-EUR",
    "BaseSymbol": {
      "Value": "ADA"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "AE-BTC",
    "BaseSymbol": {
      "Value": "AE"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "AE-EUR",
    "BaseSymbol": {
      "Value": "AE"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "AION-BTC",
    "BaseSymbol": {
      "Value": "AION"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "AION-EUR",
    "BaseSymbol": {
      "Value": "AION"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ANT-BTC",
    "BaseSymbol": {
      "Value": "ANT"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ANT-EUR",
    "BaseSymbol": {
      "Value": "ANT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ARK-BTC",
    "BaseSymbol": {
      "Value": "ARK"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ARK-EUR",
    "BaseSymbol": {
      "Value": "ARK"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "BAL-EUR",
    "BaseSymbol": {
      "Value": "BAL"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "BAND-EUR",
    "BaseSymbol": {
      "Value": "BAND"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "BAT-BTC",
    "BaseSymbol": {
      "Value": "BAT"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "BAT-EUR",
    "BaseSymbol": {
      "Value": "BAT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "BCH-BTC",
    "BaseSymbol": {
      "Value": "BCH"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "BCH-EUR",
    "BaseSymbol": {
      "Value": "BCH"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "BNB-EUR",
    "BaseSymbol": {
      "Value": "BNB"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "BSV-BTC",
    "BaseSymbol": {
      "Value": "BSV"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "BSV-EUR",
    "BaseSymbol": {
      "Value": "BSV"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "BTC-EUR",
    "BaseSymbol": {
      "Value": "BTC"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "CHZ-EUR",
    "BaseSymbol": {
      "Value": "CHZ"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "CMT-BTC",
    "BaseSymbol": {
      "Value": "CMT"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "CMT-EUR",
    "BaseSymbol": {
      "Value": "CMT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "COMP-EUR",
    "BaseSymbol": {
      "Value": "COMP"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "DAI-EUR",
    "BaseSymbol": {
      "Value": "DAI"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "DCR-BTC",
    "BaseSymbol": {
      "Value": "DCR"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "DCR-EUR",
    "BaseSymbol": {
      "Value": "DCR"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "DGB-BTC",
    "BaseSymbol": {
      "Value": "DGB"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "DGB-EUR",
    "BaseSymbol": {
      "Value": "DGB"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "DOGE-EUR",
    "BaseSymbol": {
      "Value": "DOGE"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "DOT-EUR",
    "BaseSymbol": {
      "Value": "DOT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ELF-BTC",
    "BaseSymbol": {
      "Value": "ELF"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ELF-EUR",
    "BaseSymbol": {
      "Value": "ELF"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ENJ-BTC",
    "BaseSymbol": {
      "Value": "ENJ"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ENJ-EUR",
    "BaseSymbol": {
      "Value": "ENJ"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "EOS-BTC",
    "BaseSymbol": {
      "Value": "EOS"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "EOS-EUR",
    "BaseSymbol": {
      "Value": "EOS"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ETC-BTC",
    "BaseSymbol": {
      "Value": "ETC"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ETC-EUR",
    "BaseSymbol": {
      "Value": "ETC"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ETH-BTC",
    "BaseSymbol": {
      "Value": "ETH"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ETH-EUR",
    "BaseSymbol": {
      "Value": "ETH"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "FET-EUR",
    "BaseSymbol": {
      "Value": "FET"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "GAS-BTC",
    "BaseSymbol": {
      "Value": "GAS"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "GAS-EUR",
    "BaseSymbol": {
      "Value": "GAS"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "GLM-EUR",
    "BaseSymbol": {
      "Value": "GLM"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "GNO-EUR",
    "BaseSymbol": {
      "Value": "GNO"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "GNT-BTC",
    "BaseSymbol": {
      "Value": "GNT"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "GNT-EUR",
    "BaseSymbol": {
      "Value": "GNT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "HOT-BTC",
    "BaseSymbol": {
      "Value": "HOT"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "HOT-EUR",
    "BaseSymbol": {
      "Value": "HOT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ICX-BTC",
    "BaseSymbol": {
      "Value": "ICX"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ICX-EUR",
    "BaseSymbol": {
      "Value": "ICX"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "IOST-BTC",
    "BaseSymbol": {
      "Value": "IOST"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "IOST-EUR",
    "BaseSymbol": {
      "Value": "IOST"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "KMD-BTC",
    "BaseSymbol": {
      "Value": "KMD"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "KMD-EUR",
    "BaseSymbol": {
      "Value": "KMD"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "KNC-EUR",
    "BaseSymbol": {
      "Value": "KNC"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "LINK-BTC",
    "BaseSymbol": {
      "Value": "LINK"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "LINK-EUR",
    "BaseSymbol": {
      "Value": "LINK"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "LOOM-EUR",
    "BaseSymbol": {
      "Value": "LOOM"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "LRC-BTC",
    "BaseSymbol": {
      "Value": "LRC"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "LRC-EUR",
    "BaseSymbol": {
      "Value": "LRC"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "LSK-BTC",
    "BaseSymbol": {
      "Value": "LSK"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "LSK-EUR",
    "BaseSymbol": {
      "Value": "LSK"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "LTC-BTC",
    "BaseSymbol": {
      "Value": "LTC"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "LTC-EUR",
    "BaseSymbol": {
      "Value": "LTC"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "MANA-EUR",
    "BaseSymbol": {
      "Value": "MANA"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "MATIC-EUR",
    "BaseSymbol": {
      "Value": "MATIC"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "MIOTA-BTC",
    "BaseSymbol": {
      "Value": "MIOTA"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "MIOTA-EUR",
    "BaseSymbol": {
      "Value": "MIOTA"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "MKR-EUR",
    "BaseSymbol": {
      "Value": "MKR"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "NANO-BTC",
    "BaseSymbol": {
      "Value": "NANO"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "NANO-EUR",
    "BaseSymbol": {
      "Value": "NANO"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "NAS-BTC",
    "BaseSymbol": {
      "Value": "NAS"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "NAS-EUR",
    "BaseSymbol": {
      "Value": "NAS"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "NEO-BTC",
    "BaseSymbol": {
      "Value": "NEO"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "NEO-EUR",
    "BaseSymbol": {
      "Value": "NEO"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "NMR-EUR",
    "BaseSymbol": {
      "Value": "NMR"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "NPXS-BTC",
    "BaseSymbol": {
      "Value": "NPXS"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "NPXS-EUR",
    "BaseSymbol": {
      "Value": "NPXS"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "NULS-BTC",
    "BaseSymbol": {
      "Value": "NULS"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "NULS-EUR",
    "BaseSymbol": {
      "Value": "NULS"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "OGN-EUR",
    "BaseSymbol": {
      "Value": "OGN"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "OMG-BTC",
    "BaseSymbol": {
      "Value": "OMG"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "OMG-EUR",
    "BaseSymbol": {
      "Value": "OMG"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ONG-BTC",
    "BaseSymbol": {
      "Value": "ONG"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ONG-EUR",
    "BaseSymbol": {
      "Value": "ONG"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ONT-BTC",
    "BaseSymbol": {
      "Value": "ONT"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ONT-EUR",
    "BaseSymbol": {
      "Value": "ONT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "OXT-EUR",
    "BaseSymbol": {
      "Value": "OXT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "POWR-BTC",
    "BaseSymbol": {
      "Value": "POWR"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "POWR-EUR",
    "BaseSymbol": {
      "Value": "POWR"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "PUNDIX-EUR",
    "BaseSymbol": {
      "Value": "PUNDIX"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "QTUM-BTC",
    "BaseSymbol": {
      "Value": "QTUM"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "QTUM-EUR",
    "BaseSymbol": {
      "Value": "QTUM"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "RDD-BTC",
    "BaseSymbol": {
      "Value": "RDD"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "RDD-EUR",
    "BaseSymbol": {
      "Value": "RDD"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "REN-EUR",
    "BaseSymbol": {
      "Value": "REN"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "REP-EUR",
    "BaseSymbol": {
      "Value": "REP"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "REQ-BTC",
    "BaseSymbol": {
      "Value": "REQ"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "REQ-EUR",
    "BaseSymbol": {
      "Value": "REQ"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "RLC-EUR",
    "BaseSymbol": {
      "Value": "RLC"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "RSR-EUR",
    "BaseSymbol": {
      "Value": "RSR"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "RVN-BTC",
    "BaseSymbol": {
      "Value": "RVN"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "RVN-EUR",
    "BaseSymbol": {
      "Value": "RVN"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "SNT-BTC",
    "BaseSymbol": {
      "Value": "SNT"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "SNT-EUR",
    "BaseSymbol": {
      "Value": "SNT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "SNX-EUR",
    "BaseSymbol": {
      "Value": "SNX"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "SOL-EUR",
    "BaseSymbol": {
      "Value": "SOL"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "STEEM-BTC",
    "BaseSymbol": {
      "Value": "STEEM"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "STEEM-EUR",
    "BaseSymbol": {
      "Value": "STEEM"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "STMX-BTC",
    "BaseSymbol": {
      "Value": "STMX"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "STMX-EUR",
    "BaseSymbol": {
      "Value": "STMX"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "STORJ-EUR",
    "BaseSymbol": {
      "Value": "STORJ"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "STORM-BTC",
    "BaseSymbol": {
      "Value": "STORM"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "STORM-EUR",
    "BaseSymbol": {
      "Value": "STORM"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "STRAT-BTC",
    "BaseSymbol": {
      "Value": "STRAT"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "STRAT-EUR",
    "BaseSymbol": {
      "Value": "STRAT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "STRAX-EUR",
    "BaseSymbol": {
      "Value": "STRAX"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "SUSHI-EUR",
    "BaseSymbol": {
      "Value": "SUSHI"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "SXP-EUR",
    "BaseSymbol": {
      "Value": "SXP"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "TRB-EUR",
    "BaseSymbol": {
      "Value": "TRB"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "TRX-BTC",
    "BaseSymbol": {
      "Value": "TRX"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "TRX-EUR",
    "BaseSymbol": {
      "Value": "TRX"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "UMA-EUR",
    "BaseSymbol": {
      "Value": "UMA"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "UNI-EUR",
    "BaseSymbol": {
      "Value": "UNI"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "USDC-EUR",
    "BaseSymbol": {
      "Value": "USDC"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "USDT-EUR",
    "BaseSymbol": {
      "Value": "USDT"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "UTK-EUR",
    "BaseSymbol": {
      "Value": "UTK"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "VET-BTC",
    "BaseSymbol": {
      "Value": "VET"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "VET-EUR",
    "BaseSymbol": {
      "Value": "VET"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "VTC-BTC",
    "BaseSymbol": {
      "Value": "VTC"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "VTC-EUR",
    "BaseSymbol": {
      "Value": "VTC"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "VTHO-BTC",
    "BaseSymbol": {
      "Value": "VTHO"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "VTHO-EUR",
    "BaseSymbol": {
      "Value": "VTHO"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "WAVES-BTC",
    "BaseSymbol": {
      "Value": "WAVES"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "WAVES-EUR",
    "BaseSymbol": {
      "Value": "WAVES"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "WTC-BTC",
    "BaseSymbol": {
      "Value": "WTC"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "WTC-EUR",
    "BaseSymbol": {
      "Value": "WTC"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "XEM-BTC",
    "BaseSymbol": {
      "Value": "XEM"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "XEM-EUR",
    "BaseSymbol": {
      "Value": "XEM"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "XLM-BTC",
    "BaseSymbol": {
      "Value": "XLM"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "XLM-EUR",
    "BaseSymbol": {
      "Value": "XLM"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "XRP-BTC",
    "BaseSymbol": {
      "Value": "XRP"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "XRP-EUR",
    "BaseSymbol": {
      "Value": "XRP"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "XTZ-BTC",
    "BaseSymbol": {
      "Value": "XTZ"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "XTZ-EUR",
    "BaseSymbol": {
      "Value": "XTZ"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "XVG-BTC",
    "BaseSymbol": {
      "Value": "XVG"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "XVG-EUR",
    "BaseSymbol": {
      "Value": "XVG"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "YFI-EUR",
    "BaseSymbol": {
      "Value": "YFI"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "YFII-EUR",
    "BaseSymbol": {
      "Value": "YFII"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ZIL-BTC",
    "BaseSymbol": {
      "Value": "ZIL"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ZIL-EUR",
    "BaseSymbol": {
      "Value": "ZIL"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  },
  {
    "Name": "ZRX-BTC",
    "BaseSymbol": {
      "Value": "ZRX"
    },
    "QuoteSymbol": {
      "Value": "BTC"
    }
  },
  {
    "Name": "ZRX-EUR",
    "BaseSymbol": {
      "Value": "ZRX"
    },
    "QuoteSymbol": {
      "Value": "EUR"
    }
  }
]
